src/go/rpk/pkg/cli/topic/produce.go

```diff
@@ -107,6 +116,18 @@ func newProduceCommand(fs afero.Fs, p *config.Params) *cobra.Command {
 				out.Die("invalid empty format")
 			}
 
+			isSchemaRegistry := schemaID > 0 || keySchemaID > 0
```

Comment:
>= 0, rather than > 0? I don't think anything forbids a schema from having an ID of 0

Reply:
Fixed in https://github.com/redpanda-data/redpanda/compare/291bc40a6f826619022eafc0dcd87a2ee35359c2..5729c8ba8d0b43f6c74eaf7c50182fed8c00f450